src/CTA.Rules.Config/Utils.cs

```diff
@@ -267,5 +275,27 @@ public static void DownloadFilesToFolder(string s3Bucket, string targetFolder, L
                 }
             });
         }
+
+        public static IEnumerable<string> GetProjectPaths(string solutionPath)
+        {
+            if(solutionPath.Contains(".sln") && File.Exists(solutionPath))
+            {
+                string solutionDir = Directory.GetParent(solutionPath).FullName;
+                IEnumerable<string> projectPaths = null;
+                try
+                {
+                    SolutionFile solution = SolutionFile.Parse(solutionPath);
+                    projectPaths = solution.ProjectsInOrder.Where(p => AcceptedProjectTypes.Contains(p.ProjectType)).Select(p => p.AbsolutePath);
+                }
+                catch(Exception ex)
+                {
+                    LogHelper.LogError(ex, $"Error while parsing solution file {solutionPath} falling back to directory parsing.");
+                    projectPaths = Directory.EnumerateFiles(solutionDir, "*.csproj", SearchOption.AllDirectories);
+                }
+
+                return projectPaths;
+            }
+            return null;
+        }
     }
```

Comment:
what are your thoughts on moving this to codelyzer so we can use in all the projects using it?

Reply:
Agreed, moving to codelyzer, removing project types to consume all but move pruning logic to build function to only attempt to build projects of certain types.